### ISSUE TITLE
TabMon Service won't start and stay running

### ISSUE THREAD

**Issue by TabMonDBA** [tableau/TabMon#30]
I ran the TabMon service manually by pasting the 'Path to Executable' command into a command window and this a sample of the errors - see below " Cannot add non-nullable columns to an existing database table!" Therefore the TABMON database is not retrieving any data currently.  The row count is same as last week.

Any help would be greatly appreciated.

------------------------------------------------------------------------------------------------------------------------
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Could not communicate with Perfmon on target host 'erp-win-s27': Access is denied
Loaded 0 Perfmon counters on erp-win-s27.
Loaded 74 MBean counters on 0058-win-s205.
Loaded 58 MBean counters on 0058-win-s216.
Loaded 0 MBean counters on 0058-win-s206.
Loaded 0 MBean counters on 0058-win-s207.
Loaded 0 MBean counters on 0058-win-s209.
Loaded 0 MBean counters on 0058-win-s210.
Loaded 0 MBean counters on 0058-win-s212.
Loaded 0 MBean counters on 0058-win-s213.
Loaded 0 MBean counters on 0058-win-s214.
Loaded 0 MBean counters on 0058-win-s215.
Loaded 0 MBean counters on 0058-win-s66.
Loaded 0 MBean counters on erp-win-s28.
Loaded 0 MBean counters on 0058-win-s139.
Loaded 0 MBean counters on erp-win-s29.
Loaded 0 MBean counters on 0058-win-s73.
Loaded 0 MBean counters on 0058-win-s61.
Loaded 0 MBean counters on 0058-win-s63.
Loaded 0 MBean counters on 0058-win-s68.
Loaded 0 MBean counters on 0058-win-s69.
Loaded 0 MBean counters on erp-win-s30.
Loaded 0 MBean counters on erp-win-s23.
Loaded 0 MBean counters on erp-win-s24.
Loaded 0 MBean counters on erp-win-s25.
Loaded 0 MBean counters on erp-win-s26.
Loaded 0 MBean counters on erp-win-s27.
TabMon initialized!  Starting performance counter polling..
The TabMon service is now running, press Control+C to exit.
Polling..
Finished polling 132 counters. [58 failures]
The service threw an unhandled exception
System.ArgumentException: Cannot add non-nullable columns to an existing database table!
Parameter name: column
   at DataTableWriter.Adapters.DbAdapter.AddColumnsToTableToMatchSchema(String t
ableName, DataTable schema)
   at DataTableWriter.DbTableManager.UpdateTableToMatchSchema(IDbAdapter adapter
, DataTable schema)
   at DataTableWriter.DbTableManager.InitializeTable(IDbAdapter adapter, DataTab
le schema, DbTableInitializationOptions tableInitializationOptions)
   at DataTableWriter.Writers.DataTableDbWriter.Write(DataTable table)
   at TabMon.TabMonAgent.Poll(Object stateInfo)
   at System.Threading.TimerQueueTimer.CallCallbackInContext(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionCo
ntext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, C
ontextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.TimerQueueTimer.CallCallback()
   at System.Threading.TimerQueueTimer.Fire()
   at System.Threading.TimerQueue.FireNextTimers()
   at System.Threading.TimerQueue.AppDomainTimerCallback()

**Comment by jmangue** [comment]
Hi!  What does the schema of your existing countersamples table look like?  That error message typically indicates that the table TabMon is using is missing columns that are expected to already be there.

One thing to try is opening your TabMon config and try changing the table name (<Table name="countersamples"/>)  to a new table, then restart TabMon.  This will cause a new output table to be generated at startup and should fix your issue.

**Comment by jmangue** [comment]
Eeek, github stripped the xml out of my comment... but what I was trying to convey is that the table name is kept in a "Table" element within the "Database" element in TabMon.config.

**Comment by jmangue** [comment]
Closing out as inactive, please reopen if you'd like more help on this issue.

**Comment by 11kasper11** [comment]
Hi,
I have the same error message in TabMon, any idea what can be the reason? 
We have installed and running TabMon for few years, but this summer it failed with error "Cannot add non-nullable columns to an existing database table!" on two environments at the same time. Resolution was rename countersamples table name in config, new table works! I compared columns in old and new table and everything seems to be the same, so I don´t understand why is this issue... 
Thanks :-)
